torch/nn/modules/module.py

```diff
@@ -577,21 +577,24 @@ def __setstate__(self, state):
         if '_load_state_dict_pre_hooks' not in self.__dict__:
             self._load_state_dict_pre_hooks = OrderedDict()
 
-    def __getattr__(self, name):
-        if '_parameters' in self.__dict__:
-            _parameters = self.__dict__['_parameters']
-            if name in _parameters:
-                return _parameters[name]
-        if '_buffers' in self.__dict__:
-            _buffers = self.__dict__['_buffers']
-            if name in _buffers:
-                return _buffers[name]
-        if '_modules' in self.__dict__:
-            modules = self.__dict__['_modules']
-            if name in modules:
-                return modules[name]
-        raise AttributeError("'{}' object has no attribute '{}'".format(
-            type(self).__name__, name))
+    def __getattribute__(self, name):
+        try:
+            return super(Module, self).__getattribute__(name)
+        except AttributeError as err:
+            if '_parameters' in self.__dict__:
```

Comment:
I'd feel better (and it would be faster) if the `self` accesses here were done using `super(Module, self).__getattribute__`. As it stands right now, you are redispatching the `self.__dict__` lookup to `__getattribute__`, which eventually works but the second round through getattribute is totally unnecessary.

Also, it's pretty weird to define it this way, so I'd definitely recommend having a comment explaining why we do it this way.

Also, needs a test!